source/common/router/config_impl.cc

```diff
@@ -431,6 +432,13 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
     regex_rewrite_ = Regex::Utility::parseRegex(rewrite_spec.pattern());
     regex_rewrite_substitution_ = rewrite_spec.substitution();
   }
+
+  if (path_redirect_has_query_ && strip_query_) {
+    ENVOY_LOG(warn,
+              "`strip_query` is set to true, but `path_redirect` contains query string and it will "
+              "not be stripped: {}",
+              path_redirect_);
+  }
```

Comment:
If we have a runtime guard do we need this warning? Should this potentially just be a failure? Again I'm confused as to the behavior.